examples/knative_postgresql_customvar/Makefile

```diff
@@ -0,0 +1,180 @@
+
+.PHONY: install-backing-db-operator
+## Install the Backing Service DB Operator
+install-backing-db-operator:
+	@. ../../hack/examples-commons.sh && \
+	install_postgresql_operator_source && \
+	install_postgresql_operator_subscription
+
+.PHONY: uninstall-backing-db-operator
+## Uninstall the Backing Service DB Operator
+uninstall-backing-db-operator:
+	@-. ../../hack/examples-commons.sh && \
+	uninstall_postgresql_operator_subscription && \
+	uninstall_postgresql_operator_source
+
+.PHONY: install-serverless-operator
+## Install the Serverless Operator
+install-serverless-operator:
+	@. ../../hack/examples-commons.sh && \
+	install_serverless_operator_subscription
+
+.PHONY: uninstall-serverless-operator
+## Uninstall the Serverless Operator
+uninstall-serverless-operator:
+	@-. ../../hack/examples-commons.sh && \
+	uninstall_serverless_operator_subscription
+
+.PHONY: install-service-mesh-operator
+## Install the Service Mesh Operator
+install-service-mesh-operator:
+	@. ../../hack/examples-commons.sh && \
+	install_service_mesh_operator_subscription
+
+.PHONY: uninstall-service-mesh-operator
+## Uninstall the Service Mesh Operator
+uninstall-service-mesh-operator:
+	@-. ../../hack/examples-commons.sh && \
+	uninstall_service_mesh_operator_subscription
+
+.PHONY: install-all-operators
+## Install all the operators
+install-all-operators: install-backing-db-operator install-service-binding-operator install-serverless-operator install-service-mesh-operator
+
+.PHONY: uninstall-all-operators
+## Uninstall all the operators
+uninstall-all-operators: uninstall-backing-db-operator uninstall-service-binding-operator uninstall-serverless-operator uninstall-service-mesh-operator
+
+.PHONY: install-all
+## Install all the operators and the quarkus native s2i builder image
+install-all: create-project install-all-operators install-quarkus-native-s2i-builder
+
+.PHONY: install-quarkus-native-s2i-builder
+## Install ubi-quarkus-native-s2i builder
+install-quarkus-native-s2i-builder:
+	@oc import-image quay.io/quarkus/ubi-quarkus-native-s2i:19.1.1 -n openshift --confirm
+	@oc patch is ubi-quarkus-native-s2i -n openshift -p '{"spec": {"tags": [{"name" : "19.1.1", "annotations": {"tags": "builder"}}]}}'
+
+## -- Application Developer targets --
+
+.PHONY: create-project
+## Create the OpenShift project/namespace
+create-project:
+	@. ../../hack/examples-commons.sh && \
+	create_project
+
+.PHONY: delete-project
+## Delete the OpenShift project/namespace
+delete-project:
+	@. ../../hack/examples-commons.sh && \
+	delete_project
+
+
+.PHONY: create-backing-db-instance
+## Create the Backing Service Database
+create-backing-db-instance:
+	@. ../../hack/examples-commons.sh && \
+	install_postgresql_db_instance
+
+#PHONY: set-labels-on-knative-app
+## Set the binding labels for the app
+#set-labels-on-knative-app:
+#	@oc project service-binding-demo
+#	@oc patch dc knative-app -p '{"metadata": {"labels": {"connects-to": "postgres", "environment": "demo"}}}'
```

Comment:
Would we also want to create a hack script for this target? The script could take the full DeploymentConfig string as input. Or - maybe would get more clever and find the DC based on the "knative-app" substring in the DC name string?

Reply:
This target is specific for the example's application and could be (in the worst case) different for each example. So there's no need for the "hack script function" for this and the target should remain in the example's `Makefile`.

In the case of this (knative) example the `ServiceBindingRequest`'s `applicationSelector` uses the `resourceRef` field rather then `matchLabels` so set-labels-on-app target is unnecessary